spec/beaker-pe/pe-client-tools/executable_helper_spec.rb

```diff
@@ -55,4 +57,52 @@ class MixedWithExecutableHelper
   it 'has a method to login with puppet access' do
     expect(subject.respond_to?('login_with_puppet_access_on')).not_to be(false)
   end
-end
\ No newline at end of file
+
+  context 'puppet access login with lifetime parameter' do
+    let(:host) {make_host('my_super_host', {:pe_ver => '3.0',
+                                            :platform => 'linux',
+                                            :roles => ['master', 'database', 'dashboard', 'agent'],
+                                            :type => 'pe'})}
+
+    #let(:host) {Beaker::Host.create('my_super_host', 
+    #                                {:roles => ['master', 'database', 'dashboard', 'agent'],
+    #                                 :platform => 'linux'})}
+    let(:credentials) { { login: username, password: password} }
+    let(:username) {'T'}
+    let(:password) {'Swift'}
+    let(:mock_page) {double('mock_page')}
+
+    subject {Scooter::HttpDispatchers::ConsoleDispatcher.new(host.to_s, credentials)}
+    #
+    #    let(:logger) { double('logger', 
+    #                          :log_level => Logger::DEBUG,
+    #                          :step_in => false)}
+    #    unixhost = { roles:     ['test_role'],
+    #                 'platform' => 'debian-7-x86_64' }
+    #    let(:host) { Beaker::Host.create('test.com', unixhost, {:logger => logger}) }
+    before do
+      #expect(OpenSSL::PKey).to receive(:read).and_return('Pkey')
+      #expect(OpenSSL::X509::Certificate).to receive(:new).and_return('client_cert')
+      allow_any_instance_of(Scooter::HttpDispatchers::HttpDispatcher).to receive(:get_host_cert) {'host cert'}
+      allow_any_instance_of(Scooter::HttpDispatchers::HttpDispatcher).to receive(:get_host_private_key) {'key file'}
+      allow_any_instance_of(Scooter::HttpDispatchers::HttpDispatcher).to receive(:get_host_cacert) {'cert file'}
+      expect(subject).to be_kind_of(Scooter::HttpDispatchers::ConsoleDispatcher)
+      index = subject.connection.builder.handlers.index(Faraday::Adapter::NetHttp)
+      subject.connection.builder.swap(index, Faraday::Adapter::Test) do |stub|
+        head = {"server"=>"nginx/1.8.1",
+                "date"=>"Tue, 29 Nov 2016 22:05:41 GMT",
+                "content-length"=>"0",
+                "connection"=>"close",
+                "set-cookie"=>"JSESSIONID=b05e9b11-5e9f-4d6a-9faf-e28a0415197d; Path=/; Secure; HttpOnly, rememberMe=deleteMe; Path=/auth; Max-Age=0; Expires=Mon, 28-Nov-2016 22:05:41 GMT, pl_ssti=0CeHhpz5PPLna7kpaEMcTHjJ62z9eizHTzsxEXNK8W20;Secure;Path=/",
+                "location"=>"/",
+                "x-frame-options"=>"DENY"}
+        stub.post('/auth/login', "username=#{username}&password=#{password}") {[200, head, '']}
+                  stub.get('/') {[200, {}, '']}
+      end
+    end
+
+    it 'accepts correct value' do
+      expect{subject.signin}.to_not raise_error
```

Comment:
I'm not sure that `signin` will actually trigger this code path.  
I think this is what you want.  
`expect{subject.login_with_puppet_access_on(test_host, test_dispatcher, :lifetime => '5d')}.not_to raise_error`